community-tutorials/pufferpanel/01-en.md

```diff
@@ -0,0 +1,182 @@
+---
+title: Pufferpanel on a Debian/Ubuntu Server
+description: How to install the Pufferpanel on Debian or Ubuntu servers
+updated_at: 2021-11-09
+slug: install-pufferpanel-debian-ubuntu
+author_name: Jonas Löchner
+author_url: https://sancraft.dev/
+author_image: https://dl.sancraft.dev/images/Sanoj/1-B.png
+author_bio: -
+tags: [mc, minecraft, web, pufferpanel, panel, linux, debian, ubuntu]
+netcup_product_url: https://netcup.de/bestellen/produkt.php?produkt=2623
+language: en
+available_languages: en
+---
+
+# Introduction
+This tutorial will show you, how you can install the Pufferpanel on a Debian or Ubuntu Server on the server "RS 2000" trough apt and how you can set up a apache2 reverse proxy to it <br>
+The Pufferpanel is a simple Webpanel for game servers and many more! <br>
+I am doing this on a debian 11 server, but it also should work on other debian versions and on ubuntu <br>
+This Tutorial is based on the Pufferpanel Docs https://docs.pufferpanel.com/en/2.x/installing.html, on https://github.com/SanCraftDev/Debian-Setup/blob/main/README.md#Snapd and on https://github.com/SanCraftDev/Debian-Setup/blob/main/README.md#Apache-and-Certbot
+
+# Requirements
+You need a Debian or Ubuntu Server with an Internet Connection, which is accessible from the internet over a static IP on port 8080 and 5657. <br>
+You also need root permissions on your server. <br>
+If you want to use https and a domain or subdomain, you need also to have the server open to the internet on port 80 and 443
+
+# Step 1 - Installing the Panel
+First you need to connect to your server over ssh. <br>
+Then we are going to switch to the root user and are going to update the system, and then we are going to install the Pufferpanel
+```sh
+su
+apt update && apt upgrade -y && apt autoremove -y
+apt install wget curl sudo nano screen cron -y
+curl -s https://packagecloud.io/install/repositories/pufferpanel/pufferpanel/script.deb.sh | sudo bash
+sudo apt-get install pufferpanel -y
+```
+Now Create a start file:
+```sh
+nano /etc/pufferpanel/start.sh
+```
+Now paste here:
+```
+screen -r pp -X quit
+screen -AmdS pp /usr/sbin/pufferpanel run
+```
+Save your edit with ctrl + x, then press y and now save it with enter <br>
+Now run to start pufferpanel and run it on boot:
+```sh
+chmod +x /etc/pufferpanel/start.sh
+{ crontab -l 2>/dev/null; echo "@reboot sleep 10 && cd /etc/pufferpanel/ && ./start.sh" ; } | crontab -
+```
+
+# Step 1.1 - Adding a User & starting the panel
+Now we to add a user. <br>
+```sh
+pufferpanel user add
+```
+```
+First enter a Username
+Now enter your email address
+Next enter a secure password and confirm it again
+Select y to make the user a admin user
+```
+Now you can start the panel the first time
+```sh
+/etc/pufferpanel/start.sh
```

Comment:
When I run this command, I´m getting this error: `No screen session found`

Reply:
this is because, the first line of this file tried to stop the pufferpanel if it is running, if it is not running this error comes, but everything should work, but i removed this lines because it is not working